internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -395,8 +407,138 @@ func mutatingToWebhookDescription(webhook admissionregv1.MutatingWebhook, depNam
 	return description
 }
 
+// The default AdmissionReviewVersions set in a CSV for conversion webhooks
+// if ConversionReviewVersions is not specified in CRDs.
+var defaultConversionReviewVersions = []string{"v1beta1", "v1"}
+
+// conversionToWebhookDescription builds a webhookDescription for conversion webhook with the
+// WebhookClientConfig specified in either v1 or v1beta1 CRDs.
+func conversionToWebhookDescription(crdNames []string, v1WebhookClientConfig *apiextv1.WebhookConversion, v1beta1WebhookClientConfig *apiextv1beta1.CustomResourceConversion, depName string, ws *corev1.Service) operatorsv1alpha1.WebhookDescription {
+
+	conversionReviewVersions, whPath, serviceRefName, serviceRefPort := getConversionWebhookDetails(v1WebhookClientConfig, v1beta1WebhookClientConfig)
+
+	description := operatorsv1alpha1.WebhookDescription{
+		Type:                    operatorsv1alpha1.ConversionWebhook,
+		ConversionCRDs:          crdNames,
+		AdmissionReviewVersions: conversionReviewVersions,
+		WebhookPath:             &whPath,
+		DeploymentName:          depName,
+		SideEffects: func() *admissionregv1.SideEffectClass {
+			seNone := admissionregv1.SideEffectClassNone
+			return &seNone
+		}(),
+	}
+
+	if len(description.AdmissionReviewVersions) == 0 {
+		description.AdmissionReviewVersions = defaultConversionReviewVersions
+	}
+
+	var webhookServiceRefPort int32 = 443
+
+	if serviceRefPort != nil {
+		webhookServiceRefPort = *serviceRefPort
+	}
+
+	if ws != nil {
+		for _, port := range ws.Spec.Ports {
+			if webhookServiceRefPort == port.Port {
+				description.ContainerPort = port.Port
+				description.TargetPort = &port.TargetPort
+				break
+			}
+		}
+	}
+
+	if description.DeploymentName == "" {
+		description.DeploymentName = strings.TrimSuffix(serviceRefName, "-service")
+	}
+	return description
+}
+
+// getConvWebhookCRDNamesAndConfig returns the list of the CRDs which have conversion webhookClient config
+// specified. It also returns the WebhookConversion config for v1 and v1beta1 CRDs. Getting the WebhookConversion Config
+// from the CRDs is necessary to extract the serviceRef specifications for webhook.
+func getConvWebhookCRDNamesAndConfig(c *collector.Manifests, serviceName string) ([]string, *apiextv1.WebhookConversion, *apiextv1beta1.CustomResourceConversion) {
```

Comment:
This should return a `map[string]apiextv1.WebhookConversion`, where the `string` key is the CRD name. You can transform `apiextv1beta1.CustomResourceConversion` to a [`apiextv1.WebhookConversion`](https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1#WebhookConversion) like so:
```go
v1conv := apiextv1.WebhookConversion{
	ClientConfig:             &apiextv1.ClientConfig{Service: &apiextv1.ServiceReference{}},
	ConversionReviewVersions: crd.Spec.Conversion.ConversionReviewVersions,
}
if crd.Spec.Conversion.WebhookClientConfig.Service != nil {
	whConv.ClientConfig.Service.Name = crd.Spec.Conversion.WebhookClientConfig.Service.Name
	whConv.ClientConfig.Service.Port = crd.Spec.Conversion.WebhookClientConfig.Service.Port
	whConv.ClientConfig.Service.Path = crd.Spec.Conversion.WebhookClientConfig.Service.Path
}
```

Reply:
So for each `crd` present in `crdNames`, we will have a separate `webhookDefinition`? And if, again a particular CRD has both v1 and v1beta1, we will chose the config for v1CRD.

But what if the case is - multiple services serve the same conversion CRD as you mentioned before. Then again we will be overwriting the clientConfig for the same CRD.

Reply:
You can/should further de-duplicate by port+path+rewviewversions